ring_theory/subring.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2018 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
```

Comment:
Who wrote this file?

Reply:
I believe it was Johan

Reply:
I wrote this for the perfectoid project.